src/opendr/perception/object_detection_2d/yolov3/yolov3_learner.py

```diff
@@ -431,12 +431,9 @@ def infer(self, img, threshold=0.1, keep_size=True):
 
         self._model.set_nms(nms_thresh=0.45, nms_topk=400)
 
-        if isinstance(img, Image):
-            _img = img.numpy()
-        elif isinstance(img, np.ndarray):
-            _img = img
-        else:
-            raise ValueError("Input should be of type Image or numpy array.")
+        if not isinstance(img, Image):
+            img = Image(img)
```

Comment:
This doesn't reproduce the previous code.

Reply:
Fixed by improving constructor.